### PR TITLE
Narrow down RBAC rules

### PR DESCRIPTION
**Pull request by johnugeorge**
This PR narrows down the RBAC rules for Katib controller installation

Fixes: #1639

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
@johnugeorge Thanks for this!
I confirmed this RBAC on my local using the following operation:

- create and delete random.yaml by kubectl and UI.
- create and delete from-volume-resume.yaml by kubectl and UI.

That works well.

Also, maybe we want to generate RBAC manifests by `sigs.k8s.io/controller-tools/cmd/controller-gen` in the future.

> Note: Currently, cert-generator requires full access to secrets and UI needs full access to configmaps. Can we reduce it in the future?

I guess the cert-generator has limited access to secrets, now.

https://github.com/kubeflow/katib/blob/0749265a22f5fad30ac148ffc35885f47dbb5b1a/manifests/v1beta1/components/cert-generator/rbac.yaml#L12-L15

/lgtm

**Comment by johnugeorge** [comment]
> @johnugeorge Thanks for this! I confirmed this RBAC on my local using the following operation:
> 
> * create and delete random.yaml by kubectl and UI.
> * create and delete from-volume-resume.yaml by kubectl and UI.
> 
> That works well.
> 
> Also, maybe we want to generate RBAC manifests by `sigs.k8s.io/controller-tools/cmd/controller-gen` in the future.
> 
> > Note: Currently, cert-generator requires full access to secrets and UI needs full access to configmaps. Can we reduce it in the future?
> 
> I guess the cert-generator has limited access to secrets, now.
> 
> https://github.com/kubeflow/katib/blob/0749265a22f5fad30ac148ffc35885f47dbb5b1a/manifests/v1beta1/components/cert-generator/rbac.yaml#L12-L15
> 
> /lgtm

Fixed configmaps and secrets as well

**Comment by tenzen-y** [comment]
I have confirmed new changes using the same way as https://github.com/kubeflow/katib/pull/2091#pullrequestreview-1263745731, again.

/lgtm

**Comment by andreyvelich** [comment]
Thank you for this @johnugeorge!
I will check it on Monday.

File: manifests/v1beta1/components/controller/rbac.yaml [hunk 1]
```diff
@@ -7,19 +7,38 @@ rules:
   - apiGroups:
       - ""
     resources:
-      - configmaps
-      - serviceaccounts
       - services
       - events
-      - namespaces
+    verbs:
+      - "*"
```
Comment (andreyvelich) [inline manifests/v1beta1/components/controller/rbac.yaml:13]: Do we need `*` for services ?

File: manifests/v1beta1/components/controller/rbac.yaml [hunk 2]
```diff
@@ -7,19 +7,38 @@ rules:
...
       - persistentvolumeclaims
+    verbs:
+      - "get"
+      - "list"
+      - "watch"
+      - "create"
+  - apiGroups:
+      - ""
+    resources:
+      - namespaces
```
Comment (andreyvelich) [inline manifests/v1beta1/components/controller/rbac.yaml:28]: Do we need permission for namespace in the Controller @johnugeorge @tenzen-y ?

File: manifests/v1beta1/components/controller/rbac.yaml [hunk 3]
```diff
@@ -32,7 +51,10 @@ rules:
...
       - rolebindings
     verbs:
-      - "*"
+      - "get"
+      - "create"
+      - "list"
+      - "watch"
   - apiGroups:
       - batch
     resources:
```
Comment (andreyvelich) [inline manifests/v1beta1/components/controller/rbac.yaml:60]: @johnugeorge Should we limit scope for `deployment`, `batch`, and `kubeflow.org jobs` ?

File: manifests/v1beta1/components/controller/rbac.yaml (hunk 1 above)
Comment (johnugeorge) [reply manifests/v1beta1/components/controller/rbac.yaml:13]: Yes. We do it in  https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/suggestion_controller_util.go

File: manifests/v1beta1/components/controller/rbac.yaml (hunk 2 above)
Comment (johnugeorge) [reply manifests/v1beta1/components/controller/rbac.yaml:28]: Yes. It is needed only in UI.

File: manifests/v1beta1/components/controller/rbac.yaml (hunk 1 above)
Comment (johnugeorge) [reply manifests/v1beta1/components/controller/rbac.yaml:13]: We do get/create/delete/watch

File: manifests/v1beta1/components/controller/rbac.yaml (hunk 2 above)
Comment (johnugeorge) [reply manifests/v1beta1/components/controller/rbac.yaml:28]: I retested it. Currently, Katib needs get/list/watch for namespaces as well. Need to investigate separately

**Review by andreyvelich — COMMENTED** [review]
/lgtm

**Pull request merged**